### Import libraries, read in the data

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

sns.set_style('whitegrid')
pd.set_option('display.max_columns', None) # display all columns

%matplotlib inline 

In [21]:
kobe = pd.read_csv("data.csv/data.csv")

### Some basic summary/observation

In [22]:
kobe.head()

action_type combined_shot_type  game_event_id   game_id      lat  \
0          Jump Shot          Jump Shot             10  20000012  33.9723   
1          Jump Shot          Jump Shot             12  20000012  34.0443   
2          Jump Shot          Jump Shot             35  20000012  33.9093   
3          Jump Shot          Jump Shot             43  20000012  33.8693   
4  Driving Dunk Shot               Dunk            155  20000012  34.0443   

   loc_x  loc_y       lon  minutes_remaining  period  playoffs   season  \
0    167     72 -118.1028                 10       1         0  2000-01   
1   -157      0 -118.4268                 10       1         0  2000-01   
2   -101    135 -118.3708                  7       1         0  2000-01   
3    138    175 -118.1318                  6       1         0  2000-01   
4      0      0 -118.2698                  6       2         0  2000-01   

   seconds_remaining  shot_distance  shot_made_flag       shot_type  \
0                 27             18             NaN  2PT Field Goal   
1                 22             15               0  2PT Field Goal   
2                 45             16               1  2PT Field Goal   
3                 52             22               0  2PT Field Goal   
4                 19              0               1  2PT Field Goal   

          shot_zone_area  shot_zone_basic  shot_zone_range     team_id  \
0          Right Side(R)        Mid-Range        16-24 ft.  1610612747   
1           Left Side(L)        Mid-Range         8-16 ft.  1610612747   
2   Left Side Center(LC)        Mid-Range        16-24 ft.  1610612747   
3  Right Side Center(RC)        Mid-Range        16-24 ft.  1610612747   
4              Center(C)  Restricted Area  Less Than 8 ft.  1610612747   

            team_name   game_date    matchup opponent  shot_id  
0  Los Angeles Lakers  2000-10-31  LAL @ POR      POR        1  
1  Los Angeles Lakers  2000-10-31  LAL @ POR      POR        2  
2  Los Angeles Lakers  2000-10-31  LAL @ POR      POR        3  
3  Los Angeles Lakers  2000-10-31  LAL @ POR      POR        4  
4  Los Angeles Lakers  2000-10-31  LAL @ POR      POR        5

We don't need any information about Kobe's team since he only plays for the Lakers and the matchup information is captured by "opponent"

In [23]:
kobe.set_index('shot_id', inplace=True)
kobe.drop(['team_id', 'team_name', 'matchup'], axis=1, inplace=True)

In [24]:
kobe.head()

action_type combined_shot_type  game_event_id   game_id  \
shot_id                                                                  
1                Jump Shot          Jump Shot             10  20000012   
2                Jump Shot          Jump Shot             12  20000012   
3                Jump Shot          Jump Shot             35  20000012   
4                Jump Shot          Jump Shot             43  20000012   
5        Driving Dunk Shot               Dunk            155  20000012   

             lat  loc_x  loc_y       lon  minutes_remaining  period  playoffs  \
shot_id                                                                         
1        33.9723    167     72 -118.1028                 10       1         0   
2        34.0443   -157      0 -118.4268                 10       1         0   
3        33.9093   -101    135 -118.3708                  7       1         0   
4        33.8693    138    175 -118.1318                  6       1         0   
5        34.0443      0      0 -118.2698                  6       2         0   

          season  seconds_remaining  shot_distance  shot_made_flag  \
shot_id                                                              
1        2000-01                 27             18             NaN   
2        2000-01                 22             15               0   
3        2000-01                 45             16               1   
4        2000-01                 52             22               0   
5        2000-01                 19              0               1   

              shot_type         shot_zone_area  shot_zone_basic  \
shot_id                                                           
1        2PT Field Goal          Right Side(R)        Mid-Range   
2        2PT Field Goal           Left Side(L)        Mid-Range   
3        2PT Field Goal   Left Side Center(LC)        Mid-Range   
4        2PT Field Goal  Right Side Center(RC)        Mid-Range   
5        2PT Field Goal              Center(C)  Restricted Area   

         shot_zone_range   game_date opponent  
shot_id                                        
1              16-24 ft.  2000-10-31      POR  
2               8-16 ft.  2000-10-31      POR  
3              16-24 ft.  2000-10-31      POR  
4              16-24 ft.  2000-10-31      POR  
5        Less Than 8 ft.  2000-10-31      POR

In [25]:
kobe.dtypes

action_type            object
combined_shot_type     object
game_event_id           int64
game_id                 int64
lat                   float64
loc_x                   int64
loc_y                   int64
lon                   float64
minutes_remaining       int64
period                  int64
playoffs                int64
season                 object
seconds_remaining       int64
shot_distance           int64
shot_made_flag        float64
shot_type              object
shot_zone_area         object
shot_zone_basic        object
shot_zone_range        object
game_date              object
opponent               object
dtype: object

Describe continuous variables

In [16]:
kobe.describe(include=['number'])

game_event_id          game_id           lat         loc_x  \
count   30697.000000     30697.000000  30697.000000  30697.000000   
mean      249.190800  24764065.868065     33.953192      7.110499   
std       150.003712   7755174.890198      0.087791    110.124578   
min         2.000000  20000012.000000     33.253300   -250.000000   
25%       110.000000  20500077.000000     33.884300    -68.000000   
50%       253.000000  20900354.000000     33.970300      0.000000   
75%       368.000000  29600474.000000     34.040300     95.000000   
max       659.000000  49900088.000000     34.088300    248.000000   

              loc_y           lon  minutes_remaining        period  \
count  30697.000000  30697.000000       30697.000000  30697.000000   
mean      91.107535   -118.262690           4.885624      2.519432   
std       87.791361      0.110125           3.449897      1.153665   
min      -44.000000   -118.519800           0.000000      1.000000   
25%        4.000000   -118.337800           2.000000      1.000000   
50%       74.000000   -118.269800           5.000000      3.000000   
75%      160.000000   -118.174800           8.000000      3.000000   
max      791.000000   -118.021800          11.000000      7.000000   

           playoffs  seconds_remaining  shot_distance  shot_made_flag  \
count  30697.000000       30697.000000   30697.000000    25697.000000   
mean       0.146562          28.365085      13.437437        0.446161   
std        0.353674          17.478949       9.374189        0.497103   
min        0.000000           0.000000       0.000000        0.000000   
25%        0.000000          13.000000       5.000000        0.000000   
50%        0.000000          28.000000      15.000000        0.000000   
75%        0.000000          43.000000      21.000000        1.000000   
max        1.000000          59.000000      79.000000        1.000000   

          team_id  
count       30697  
mean   1610612747  
std             0  
min    1610612747  
25%    1610612747  
50%    1610612747  
75%    1610612747  
max    1610612747

Describe factor variables

In [17]:
kobe.describe(include=['object', 'category'])

action_type combined_shot_type   season       shot_type shot_zone_area  \
count        30697              30697    30697           30697          30697   
unique          57                  6       20               2              6   
top      Jump Shot          Jump Shot  2005-06  2PT Field Goal      Center(C)   
freq         18880              23485     2318           24271          13455   

       shot_zone_basic  shot_zone_range           team_name   game_date  \
count            30697            30697               30697       30697   
unique               7                5                   1        1559   
top          Mid-Range  Less Than 8 ft.  Los Angeles Lakers  2016-04-13   
freq             12625             9398               30697          50   

          matchup opponent  
count       30697    30697  
unique         74       33  
top     LAL @ SAS      SAS  
freq         1020     1978

### Curious about action_type and combined_shot_type variables

In [7]:
np.unique(kobe['action_type'])

array(['Alley Oop Dunk Shot', 'Alley Oop Layup shot',
       'Cutting Finger Roll Layup Shot', 'Cutting Layup Shot',
       'Driving Bank shot', 'Driving Dunk Shot',
       'Driving Finger Roll Layup Shot', 'Driving Finger Roll Shot',
       'Driving Floating Bank Jump Shot', 'Driving Floating Jump Shot',
       'Driving Hook Shot', 'Driving Jump shot', 'Driving Layup Shot',
       'Driving Reverse Layup Shot', 'Driving Slam Dunk Shot', 'Dunk Shot',
       'Fadeaway Bank shot', 'Fadeaway Jump Shot',
       'Finger Roll Layup Shot', 'Finger Roll Shot', 'Floating Jump shot',
       'Follow Up Dunk Shot', 'Hook Bank Shot', 'Hook Shot',
       'Jump Bank Shot', 'Jump Hook Shot', 'Jump Shot', 'Layup Shot',
       'Pullup Bank shot', 'Pullup Jump shot', 'Putback Dunk Shot',
       'Putback Layup Shot', 'Putback Slam Dunk Shot', 'Reverse Dunk Shot',
       'Reverse Layup Shot', 'Reverse Slam Dunk Shot', 'Running Bank shot',
       'Running Dunk Shot', 'Running Finger Roll Layup Shot',
       

In [8]:
action_type_shots_made = kobe.groupby('action_type').count()['shot_made_flag']
action_type_shots_made

action_type
Alley Oop Dunk Shot                      95
Alley Oop Layup shot                     67
Cutting Finger Roll Layup Shot            0
Cutting Layup Shot                        6
Driving Bank shot                         3
Driving Dunk Shot                       257
Driving Finger Roll Layup Shot           59
Driving Finger Roll Shot                 68
Driving Floating Bank Jump Shot           1
Driving Floating Jump Shot                3
Driving Hook Shot                        13
Driving Jump shot                        23
Driving Layup Shot                     1628
Driving Reverse Layup Shot               83
Driving Slam Dunk Shot                   43
Dunk Shot                               217
Fadeaway Bank shot                       27
Fadeaway Jump Shot                      872
Finger Roll Layup Shot                   28
Finger Roll Shot                         26
Floating Jump shot                       93
Follow Up Dunk Shot                      10
Hook Bank Shot      

### some observations
1. There are 30697 shots reorded in 1559 games
2. minutes_remaining refers to minutes remaining in a period
3. we can ignore team_id and team_name since he only played for the lakers